File: Crio/crio_programming_interview_problems-master/SortColors/SortColors.cpp
#include "../crio/cpp/io/PrintMatrix.hpp"
#include "../crio/cpp/io/ReadMatrix.hpp"
#include <bits/stdc++.h>
using namespace std;

class SortColors {
  public:
    // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    void swap(int i, int j, vector<int>& nums) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
    // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

    void sortColors(vector<int>& nums) {
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        int n = nums.size();
        int start = 0, end = n - 1;
        for (int i = 0; i < n;) {
            if (nums[i] == 2 && end > i) {
                swap(i, end, nums);
                end--;
            } else if (nums[i] == 0 && start < i) {
                swap(i, start, nums);
                start++;
            } else {
                i++;
            }
        }
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    }
};

int main() {
    int n;
    cin >> n;
    vector<int> nums;
    ReadMatrix<int>().OneDMatrix(n, nums);
    SortColors().sortColors(nums);
    PrintMatrix<int>().OneDMatrix(nums, " ");
    return 0;
}

File: Crio/crio_programming_interview_problems-master/SortColors/SortColors.py
from crio.io import PrintMatrix

def sortColors(nums):
    # CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    nums1,nums2,nums0 = 0,0,0
    for i in nums:
        if i == 1:
            nums1+=1
        elif i == 2:
            nums2+=1
        elif i == 0:
            nums0+=1
    lis = [0]*nums0 + [1]*nums1 + [2]*nums2
    return lis
    # CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

if __name__ == '__main__':
    n = int(input())
    nums = input().split()
    nums = [int(i) for i in nums]
    nums = sortColors(nums)
    PrintMatrix.OneDMatrix(nums, " ")
    